keps/sig-instrumentation/1668-trace-popagating/README.md

```diff
@@ -0,0 +1,501 @@
+
+* **Does enabling the feature change any default behavior?**
+
+  In apiserver, new request handlers added by this feature will generate
+  or update the trace context , then the trace context will be added to the
+  object's annotation by the webhook provided by this feature.
+
+  In controller-manager, when sending request to apiserver, it will get the
+  trace context from the referenced object's annotation and inject the trace context into the
+  outgoing request header with the W3C format.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+Yes
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  Objects created during the rollback will have no trace context until they
+  are recreated.
+
+* **Are there any tests for feature enablement/disablement?**
+  Unit test can ensure that the feature enablement/disablement is valid
+
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can a rollout fail? Can it impact already running workloads?**
+   It will not have impact  on running workloads.
+
+* **What specific metrics should inform a rollback?**
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+  Describe manual testing that was done and the outcomes.
+  Longer term, we may want to require automated upgrade/rollback tests, but we
+  are missing a bunch of machinery and tooling and can't do that now.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs, 
+fields of API types, flags, etc.?**
+  Even if applying deprecation policies, they may still surprise some users.
+
+### Monitoring Requirements
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can an operator determine if the feature is in use by workloads?**
+  Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+  checking if there are objects with field X set) may be a last resort. Avoid
+  logs or events for this purpose.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine 
+the health of the service?**
+  - [ ] Metrics
+    - Metric name:
+    - [Optional] Aggregation method:
+    - Components exposing the metric:
+  - [ ] Other (treat as last resort)
+    - Details:
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+  At a high level, this usually will be in the form of "high percentile of SLI
+  per day <= X". It's impossible to provide comprehensive guidance, but at the very
+  high level (needs more precise definitions) those may be things like:
+  - per-day percentage of API calls finishing with 5XX errors <= 1%
+  - 99% percentile over day of absolute value from (job creation time minus expected
+    job creation time) for cron job <= 10%
+  - 99,9% of /health requests per day finish with 200 code
```

Comment:
So while SLOs are probably N/A for this, we may want to at least include some guidance to check for performance regressions in various components where we turn this feature on.

Reply:
Thanks, we are trying to write it.
By the way, this section (Monitoring Requirements) is marked as 'must be completed when targeting beta graduation to a release', I wonder whether this will block our current KEP from being approved?"